src/components/Tool/GoogleImport/GoogleImport.tsx

```diff
@@ -129,41 +120,97 @@ const GoogleImport: React.FC<Props> = ({ accountListId }: Props) => {
   const { data, loading } = useGoogleContactGroupsQuery();
   const googleAccounts = data?.googleAccounts;
 
-  const account = useMemo(() => {
-    return googleAccounts?.find((account) => account?.id === selectedAccount);
-  }, [googleAccounts, selectedAccount]);
+  const selectedAccount = useMemo(() => {
+    return googleAccounts?.find(
+      (selectedAccount) => selectedAccount?.id === selectedAccountId,
+    );
+  }, [googleAccounts, selectedAccountId]);
 
   const initialValues = useMemo(() => {
     const initialGroupTags =
-      account?.contactGroups.reduce((acc, group) => {
-        if (group?.id) {acc[group?.id] = [group?.tag];}
+      selectedAccount?.contactGroups.reduce((acc, group) => {
+        if (group?.id) {
+          acc[group?.id] = [group?.tag];
+        }
         return acc;
       }, {}) || {};
     return {
       tagsForAllList: [],
       override: 'false',
-      importAll: 'false',
+      importByGroup: 'true',
       groupTags: initialGroupTags,
       groups: [] as string[],
     };
-  }, [account]);
+  }, [selectedAccount]);
 
   useEffect(() => {
-    googleAccounts?.length && setSelectedAccount(googleAccounts[0]?.id || '');
+    googleAccounts?.length && setSelectedAccountId(googleAccounts[0]?.id || '');
   }, [googleAccounts]);
 
   const handleCloseModal = () => {
     setShowModal(false);
-    window.location.href = `${process.env.SITE_URL}/accountLists/${accountListId}/tools`;
     setRedirecting(true);
+    window.location.href = `/accountLists/${accountListId}/tools`;
```

Comment:
Sorry, I didn't catch this earlier, but using  `https://nextjs.org/docs/pages/api-reference/functions/use-router#routerpush` is better, at least for navigations within MPDX. Instead of a full-page reload, Next.js is able to more seamlessly transition between pages, keeping the navbar visible without the whole page going white during the reload. I believe it has to load less data over the network too.